modules/bibupload/lib/bibupload.py

```diff
@@ -2274,11 +2276,11 @@ def _process_document_moreinfos(more_infos, docname, version, docformat, mode):
 
 ### Update functions
 
-def update_bibrec_date(now, bibrec_id, insert_mode_p, pretend=False):
+def update_bibrec_date(now, bibrec_id, insert_mode_p, original_005=None, pretend=False):
     """Update the date of the record in bibrec table """
     if insert_mode_p:
         query = """UPDATE bibrec SET creation_date=%s, modification_date=%s WHERE id=%s"""
-        params = (now, now, bibrec_id)
+        params = (original_005, now, bibrec_id)
```

Comment:
OK, though explicit conditional statement would leave no doubt on this, hence would be more readable in my eyes.

Perhaps you could enrich docstring saying that when `insert_mode_p` is True,  there should always be `original_005`... even when record does not have any original 005 to boot.  It seems that this variable name is a kind of misnomer, it might be nicer to call variables "original_005 -> record_creation_date" and "now -> record_modification_date", or somesuch, if clients are expected to send both.